src/commonMain/kotlin/exchange.dydx.abacus/processor/markets/OrderbookProcessor.kt

```diff
@@ -19,7 +33,101 @@ internal class OrderbookProcessor(parser: ParserProtocol) : BaseProcessor(parser
 
     private var lastOffset: Long = 0
 
-    internal fun subscribed(
+    fun processSubscribed(
+        existing: InternalMarketState,
+        tickSize: Double?,
+        content: IndexerOrderbookResponseObject?,
+    ): InternalMarketState {
+        existing.rawOrderbook = InternalOrderbook(
+            asks = content?.asks?.mapNotNull {
+                createTick(it)
+            },
+            bids = content?.bids?.mapNotNull {
+                createTick(it)
+            },
+        )
+        existing.consolidatedOrderbook = calculator.consolidate(
+            rawOrderbook = existing.rawOrderbook,
+        )
+        existing.groupedOrderbook = calculator.calculate(
+            rawOrderbook = existing.rawOrderbook,
+            tickSize = tickSize ?: 0.1,
+            groupingMultiplier = groupingMultiplier,
+        )
+        return existing
+    }
+
+    fun processChannelBatchData(
+        existing: InternalMarketState,
+        tickSize: Double?,
+        content: List<IndexerWsOrderbookUpdateResponse>?,
+    ): InternalMarketState {
+        content?.forEach {
+            processChannelData(existing, tickSize, it)
+        }
+        return existing
+    }
+
+    fun processGrouping(
+        existing: InternalMarketState,
+        tickSize: Double?,
+        groupingMultiplier: Int,
+    ): InternalMarketState {
+        this.groupingMultiplier = groupingMultiplier
+        existing.groupedOrderbook = calculator.calculate(
+            rawOrderbook = existing.rawOrderbook,
+            tickSize = tickSize ?: 0.1,
+            groupingMultiplier = groupingMultiplier,
+        )
+        return existing
+    }
+
+    private fun processChannelData(
+        existing: InternalMarketState,
+        tickSize: Double?,
+        content: IndexerWsOrderbookUpdateResponse?,
+    ): InternalMarketState {
+        existing.rawOrderbook = InternalOrderbook(
+            asks = processChangesBinary(
+                existing = existing.rawOrderbook?.asks,
+                changes = content?.asks,
+                ascending = true,
+            ),
+            bids = processChangesBinary(
+                existing = existing.rawOrderbook?.bids,
+                changes = content?.bids,
+                ascending = false,
+            ),
+        )
+        existing.consolidatedOrderbook = calculator.consolidate(
+            rawOrderbook = existing.rawOrderbook,
+        )
+        existing.groupedOrderbook = calculator.calculate(
+            rawOrderbook = existing.rawOrderbook,
+            tickSize = tickSize ?: 0.1,
```

Comment:
we use `0.1` as a default tickSize. Would it be useful to store as a const to reduce amount of magic numbers in this processor?

Reply:
👍  Addressed.